lib/browser/desktop-capturer.js

```diff
@@ -28,45 +28,47 @@ ipcMain.on('ELECTRON_BROWSER_DESKTOP_CAPTURER_GET_SOURCES', function (event, cap
 
   // If the WebContents is destroyed before receiving result, just remove the
   // reference from requestsQueue to make the module not send the result to it.
-  event.sender.once('destroyed', function () {
+  event.sender.once('destroyed', () => {
     request.webContents = null
+    return
   })
 })
 
-desktopCapturer.emit = function (event, name, sources) {
+desktopCapturer.emit = (event, name, sources) => {
   // Receiving sources result from main process, now send them back to renderer.
-  var handledRequest, i, len, ref, ref1, request, result, source, unhandledRequestsQueue
-  handledRequest = requestsQueue.shift(0)
-  result = (function () {
-    var i, len, results
-    results = []
-    for (i = 0, len = sources.length; i < len; i++) {
-      source = sources[i]
+  const handledRequest = requestsQueue.shift(0)
+  const handledWebContents = handledRequest.webContents
+  const unhandledRequestsQueue = []
+
+  const result = () => {
```

Comment:
Maybe it would be easier to generate `result` once as a [frozen](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/freeze) Object?